nvflare/private/fed/server/server_engine.py

```diff
@@ -254,11 +266,7 @@ def wait_for_complete(self, job_id):
             except BaseException:
                 with self.lock:
                     if job_id in self.run_processes:
-                        run_process_info = self.run_processes.pop(job_id)
-                        return_code = run_process_info[RunProcessKey.CHILD_PROCESS].poll()
-                        # if process exit but with Execution exception
-                        if return_code and return_code != 0:
-                            self.execution_exception_run_processes[job_id] = run_process_info
+                        self.run_processes.pop(job_id)
                 self.engine_info.status = MachineStatus.STOPPED
```

Comment:
The original logic covers the situation where an exception happens in the "finally" block in `nvflare/private/fed/server/server_app_runner.py`

Also, runner_process will throw exception as well: `https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/private/fed/app/server/runner_process.py#L129`

This new logic does not account for those situations.

Reply:
This already covered in ServerAppRunner.start_server_app() finally block.  No need for the old logic any more.

Reply:
rolled back the wait_for_complete logic.